### PR TITLE
bpo-30245: possible overflow when organize struct.pack_into error message

### PR DESCRIPTION
**Pull request by johanliu**
I just try to extend the huge sum from signed long to unsigned long in order to print it without showing overflowed effect. 

I suppose it works as expected since you can't make a buffer which size is larger than sys.maxsize. so as before, it make sense.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Modules/_struct.c [hunk 1]
```diff
@@ -1929,11 +1929,15 @@ s_pack_into(PyObject *self, PyObject **args, Py_ssize_t nargs, PyObject *kwnames
 
     /* Check boundaries */
     if ((buffer.len - offset) < soself->s_size) {
+
```
Comment (serhiy-storchaka) [inline Modules/_struct.c]: Please remove these empty lines. They look distracting.

File: Lib/test/test_struct.py [hunk 2]
```diff
@@ -599,6 +599,17 @@ def test_boundary_error_message_with_negative_offset(self):
                 'offset -11 out of range for 10-byte buffer'):
             struct.pack_into('<B', byte_list, -11, 123)
 
+    def test_boundary_error_message_with_large_offset(self):
+        # Test overflows cause by large offset and value size (issue 30245)
+        regex = (
+            "pack_into requires a buffer of at least " + str(sys.maxsize + 4) +
+            " bytes for packing 4 bytes at offset " + str(sys.maxsize) +
+            " \(actual buffer size is 10\)"
```
Comment (serhiy-storchaka) [inline Lib/test/test_struct.py]: Use raw string literals. Compiling `"\("` emits a deprecation warning.

You also could use classic string formatting, string's format() method, or f-string expression for composing this string.

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

File: Misc/ACKS [hunk 3]
```diff
@@ -1746,3 +1746,4 @@ evilzero
 Dhushyanth Ramasamy
 Subhendu Ghosh
 Sanjay Sundaresan
+Yuan Liu
```
Comment (serhiy-storchaka) [inline Misc/ACKS]: Names are sorted in roughly alphabetical order. Thus your name should be about letter L or Y (don't know what of names is a surname).

**Review by zhangyangyu — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by Mariatta** [comment]
@johanliu Your PR looks good. Have you signed the CLA? We can't accept the PR until it is signed.
Thank you.

**Comment by johanliu** [comment]
Hi, I have already signed the CLA but seems like the procedure not finished yet. It takes one week after I submit the application. What should I do right now on this issue? Thank you.

2017年5月27日 +0800 22:30 Mariatta <notifications@github.com>，写道：

@johanliu<https://github.com/johanliu> Your PR looks good. Have you signed the CLA? We can't accept the PR until it is signed.
Thank you.

**Comment by Mariatta** [comment]
Thanks @johanliu Looks like your CLA form has been received.
If you can resolve the conflict then I think this PR is ready for merge.

**Pull request merged**